go/vt/vtgate/planbuilder/plancontext/vschema.go

```diff
@@ -60,6 +60,8 @@ type VSchema interface {
 	// KeyspaceError returns any error in the keyspace vschema.
 	KeyspaceError(keyspace string) error
 
+	GetForeignKeyChecksState() sqlparser.FkChecksState
```

Comment:
It's very weird for the vcursor to depend on `sqlparser.FkChecksState` like this. The state of foreign keys goes beyond an AST identifier IMO and should be modeled as such.

Reply:
Yeah, I agree. Why does it live in the sqlparser package at all? AFAICS it's not really connected to the AST.

Reply:
Now that it is a `*bool` this isn't a problem anymore.